astro/src/pages/docs/[...slug].astro

```diff
@@ -12,8 +12,23 @@ export async function getStaticPaths() {
 
 const { entry } = Astro.props;
 const navContext = await getDocNavContext(entry.data.section);
-let { Content, headings = [] } = await entry.render();
-if (headings.length === 0 && entry.data.nestedHeadings && !entry.data.disableTOC) {
+let { Content, headings, remarkPluginFrontmatter = [] } = await entry.render();
+if ((entry.data.section === 'apis') &&
+  (entry.data.subcategory === 'identity providers') &&
+  (headings.length === 1) &&
+  ('idpDisplayName' in remarkPluginFrontmatter)) {
+  const globs = await Astro.glob('../../content/docs/apis/identity-providers/_oauth-idp-operations.mdx');
+  if (globs[0]) {
+    const fileHeadings = globs[0].getHeadings();
+    if (fileHeadings) {
+      for (const h of fileHeadings) {
+        h.text = h.text.replace('props.idp_display_name', remarkPluginFrontmatter.idpDisplayName);
```

Comment:
I like this in general and it is a clever workaround to the problem.

The only parts of this that I am not a huge fan of is replacing the `props.idp_display_name` as that might confuse people who edit those files as to what is actually happening, so maybe we could replace that for one other token.

This is also as written only going to work for the oauth-idp so this isn't a general solution. I think this pattern could be generalized out to support other pages. (the sdk pages come to mind as having some broken TOCs), however, I think we should prefer to structure our docs so as to not require it.

This is to say I think we can ship this as-is but we should be cognizant of the potential gotchas

Reply:
@vcampitelli can we use a different name property?

Reply:
@lyleschemmerling I may be wrong but I think the SDKs are a slightly different problem because the markdown is pulled from a remote URL, but maybe the same approach could work.

Reply:
Thanks for the feedback @lyleschemmerling  and @mooreds 

But I don't understand what you guys are asking.

I have to manually do that because Astro won't automatically replace that token, so it was actually displaying something like `Create the props.idp_display_name Identity Provider`.